mirrord/kube/src/api/kubernetes.rs

```diff
@@ -74,6 +75,23 @@ impl KubernetesAPI {
         }
         Ok(())
     }
+
+    /// Checks the [`CONTAINER_HAS_MESH_SIDECAR`] that is initialized in
+    /// [`super::container::choose_container`], and warns the user when they're running an
+    /// unsupported config.
+    #[tracing::instrument(level = "trace", ret, skip(self, progress))]
+    pub async fn detect_mesh_sidecar<P>(&self, progress: &mut P) -> Result<()>
+    where
+        P: Progress + Send + Sync,
+    {
+        if CONTAINER_HAS_MESH_SIDECAR.load(Acquire) {
+            progress
+                .warning("mesh/sidecar detected with `feature.network.incoming.mode = \"mirror\"`");
+        } else {
+            progress.success(Some("No mesh/sidecar detected."))
```

Comment:
I'll try `progress.success(None)` to see if it works, but having no `progress.success` at all for this would not show the warning (it displays just an `X` on the detecting part).

Reply:
Can't the warning be part of the parent/previous step?

Reply:
Probably, I'll throw it like that and see what happens.